### PR TITLE
Replace lambda with method reference

### PR DESCRIPTION
**Pull request by asashour**
This PR replaces lambda with method reference whenever possible.

### CODE REVIEW

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/flyway/FlywayAutoConfiguration.java [hunk 1]
```diff
@@ -536,7 +535,7 @@ public void customize(FluentConfiguration configuration) {
 			Postgresql properties = this.properties.getPostgresql();
 			PropertyMapper map = PropertyMapper.get().alwaysApplyingWhenNonNull();
 			map.from(properties::getTransactionalLock)
-				.to(extension.via((ext, transactionalLock) -> ext.setTransactionalLock(transactionalLock)));
+				.to(extension.via(PostgreSQLConfigurationExtension::setTransactionalLock));
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/flyway/FlywayAutoConfiguration.java:538]: Please revert these changes. We deliberately do not use method references when configuration Flyway to improve backwards- and forwards-compatibility (see https://github.com/spring-projects/spring-boot/issues/38164).

File: spring-boot-project/spring-boot-tools/spring-boot-maven-plugin/src/main/java/org/springframework/boot/maven/ArtifactsLibraries.java [hunk 2]
```diff
@@ -50,12 +50,7 @@ public class ArtifactsLibraries implements Libraries {
 	private static final Map<String, LibraryScope> SCOPES;
 
 	static {
-		Map<String, LibraryScope> libraryScopes = new HashMap<>();
-		libraryScopes.put(Artifact.SCOPE_COMPILE, LibraryScope.COMPILE);
-		libraryScopes.put(Artifact.SCOPE_RUNTIME, LibraryScope.RUNTIME);
-		libraryScopes.put(Artifact.SCOPE_PROVIDED, LibraryScope.PROVIDED);
-		libraryScopes.put(Artifact.SCOPE_SYSTEM, LibraryScope.PROVIDED);
-		SCOPES = Collections.unmodifiableMap(libraryScopes);
+		SCOPES = Map.of(Artifact.SCOPE_COMPILE, LibraryScope.COMPILE, Artifact.SCOPE_RUNTIME, LibraryScope.RUNTIME, Artifact.SCOPE_PROVIDED, LibraryScope.PROVIDED, Artifact.SCOPE_SYSTEM, LibraryScope.PROVIDED);
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot-tools/spring-boot-maven-plugin/src/main/java/org/springframework/boot/maven/ArtifactsLibraries.java:53]: Like https://github.com/spring-projects/spring-boot/pull/38352, I prefer this as currently written as I think it's easier to read. Please revert.

File: spring-boot-project/spring-boot/src/main/java/org/springframework/boot/ansi/AnsiPropertySource.java [hunk 3]
```diff
@@ -48,16 +48,13 @@ public class AnsiPropertySource extends PropertySource<AnsiElement> {
...
-		mappings.add(new EnumMapping<>("Ansi.", AnsiColor.class));
-		mappings.add(new EnumMapping<>("Ansi.BG_", AnsiBackground.class));
-		MAPPINGS = Collections.unmodifiableList(mappings);
+		MAPPINGS = List.of(new EnumMapping<>("AnsiStyle.", AnsiStyle.class),
+				new EnumMapping<>("AnsiColor.", AnsiColor.class),
+				new Ansi8BitColorMapping("AnsiColor.", Ansi8BitColor::foreground),
+				new EnumMapping<>("AnsiBackground.", AnsiBackground.class),
+				new Ansi8BitColorMapping("AnsiBackground.", Ansi8BitColor::background),
+				new EnumMapping<>("Ansi.", AnsiStyle.class), new EnumMapping<>("Ansi.", AnsiColor.class),
+				new EnumMapping<>("Ansi.BG_", AnsiBackground.class));
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot/src/main/java/org/springframework/boot/ansi/AnsiPropertySource.java:57]: This has been [proposed and declined](https://github.com/spring-projects/spring-boot/pull/38352) in the past. Please revert.

File: spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/ApplicationPidFileWriter.java [hunk 4]
```diff
@@ -70,20 +70,15 @@ public class ApplicationPidFileWriter implements ApplicationListener<SpringAppli
 	private static final List<Property> FILE_PROPERTIES;
 
 	static {
-		List<Property> properties = new ArrayList<>();
-		properties.add(new SpringProperty("spring.pid.", "file"));
-		properties.add(new SpringProperty("spring.", "pidfile"));
-		properties.add(new SystemProperty("PIDFILE"));
-		FILE_PROPERTIES = Collections.unmodifiableList(properties);
+		FILE_PROPERTIES = List.of(new SpringProperty("spring.pid.", "file"), new SpringProperty("spring.", "pidfile"),
+				new SystemProperty("PIDFILE"));
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/ApplicationPidFileWriter.java:74]: Similar to https://github.com/spring-projects/spring-boot/pull/38352. Please revert.

File: spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/ApplicationPidFileWriter.java [hunk 5]
```diff
@@ -70,20 +70,15 @@ public class ApplicationPidFileWriter implements ApplicationListener<SpringAppli
...
 
 	private static final List<Property> FAIL_ON_WRITE_ERROR_PROPERTIES;
 
 	static {
-		List<Property> properties = new ArrayList<>();
-		properties.add(new SpringProperty("spring.pid.", "fail-on-write-error"));
-		properties.add(new SystemProperty("PID_FAIL_ON_WRITE_ERROR"));
-		FAIL_ON_WRITE_ERROR_PROPERTIES = Collections.unmodifiableList(properties);
+		FAIL_ON_WRITE_ERROR_PROPERTIES = List.of(new SpringProperty("spring.pid.", "fail-on-write-error"),
+				new SystemProperty("PID_FAIL_ON_WRITE_ERROR"));
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/ApplicationPidFileWriter.java:81]: Similar to https://github.com/spring-projects/spring-boot/pull/38352. Please revert.

**Review by wilkinsona — CHANGES_REQUESTED** [review]

**Comment by philwebb** [comment]
When we merge this I think we should convert it into distinct commits for the method refs vs other stuff.

**Comment by asashour** [comment]
In this case, this PR would contain the method reference, and I would create a new one for the other changes, thanks.

File: spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-web-secure/src/test/java/smoketest/web/secure/SampleWebSecureApplicationTests.java [hunk 6]
```diff
@@ -96,7 +97,7 @@ static class SecurityConfiguration {
 
 		@Bean
 		SecurityFilterChain configure(HttpSecurity http) throws Exception {
-			http.csrf((csrf) -> csrf.disable());
+			http.csrf(AbstractHttpConfigurer::disable);
```
Comment (wilkinsona) [inline spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-web-secure/src/test/java/smoketest/web/secure/SampleWebSecureApplicationTests.java:100]: I think `CsrfConfigurer::disable` would read better here.

File: spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-web-secure-jdbc/src/main/java/smoketest/web/secure/jdbc/SampleWebSecureJdbcApplication.java [hunk 7]
```diff
@@ -48,7 +49,7 @@ protected static class ApplicationSecurity {
 
 		@Bean
 		SecurityFilterChain configure(HttpSecurity http) throws Exception {
-			http.csrf((csrf) -> csrf.disable());
+			http.csrf(AbstractHttpConfigurer::disable);
```
Comment (wilkinsona) [inline spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-web-secure-jdbc/src/main/java/smoketest/web/secure/jdbc/SampleWebSecureJdbcApplication.java:52]: I think `CsrfConfigurer::disable` would read better here.

File: spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-web-secure-custom/src/main/java/smoketest/web/secure/custom/SampleWebSecureCustomApplication.java [hunk 8]
```diff
@@ -45,7 +46,7 @@ protected static class ApplicationSecurity {
 
 		@Bean
 		SecurityFilterChain configure(HttpSecurity http) throws Exception {
-			http.csrf((csrf) -> csrf.disable());
+			http.csrf(AbstractHttpConfigurer::disable);
```
Comment (wilkinsona) [inline spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-web-secure-custom/src/main/java/smoketest/web/secure/custom/SampleWebSecureCustomApplication.java:49]: I think `CsrfConfigurer::disable` would read better here.

File: spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-web-method-security/src/main/java/smoketest/security/method/SampleMethodSecurityApplication.java [hunk 9]
```diff
@@ -75,7 +76,7 @@ protected static class ApplicationSecurity {
 
 		@Bean
 		SecurityFilterChain configure(HttpSecurity http) throws Exception {
-			http.csrf((csrf) -> csrf.disable());
+			http.csrf(AbstractHttpConfigurer::disable);
```
Comment (wilkinsona) [inline spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-web-method-security/src/main/java/smoketest/security/method/SampleMethodSecurityApplication.java:79]: I think `CsrfConfigurer::disable` would read better here.

File: spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-web-method-security/src/main/java/smoketest/security/method/SampleMethodSecurityApplication.java [hunk 10]
```diff
@@ -94,7 +95,7 @@ protected static class ActuatorSecurity {
 
 		@Bean
 		SecurityFilterChain actuatorSecurity(HttpSecurity http) throws Exception {
-			http.csrf((csrf) -> csrf.disable());
+			http.csrf(AbstractHttpConfigurer::disable);
```
Comment (wilkinsona) [inline spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-web-method-security/src/main/java/smoketest/security/method/SampleMethodSecurityApplication.java:98]: I think `CsrfConfigurer::disable` would read better here.

File: spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-session-redis/src/main/java/smoketest/session/redis/SecurityConfiguration.java [hunk 11]
```diff
@@ -41,7 +42,7 @@ SecurityFilterChain managementSecurityFilterChain(HttpSecurity http) throws Exce
 		});
 		http.formLogin(withDefaults());
 		http.httpBasic(withDefaults());
-		http.csrf((csrf) -> csrf.disable());
+		http.csrf(AbstractHttpConfigurer::disable);
```
Comment (wilkinsona) [inline spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-session-redis/src/main/java/smoketest/session/redis/SecurityConfiguration.java:45]: I think `CsrfConfigurer::disable` would read better here.

File: spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-session-mongo/src/main/java/smoketest/session/mongodb/SecurityConfiguration.java [hunk 12]
```diff
@@ -41,7 +42,7 @@ SecurityFilterChain managementSecurityFilterChain(HttpSecurity http) throws Exce
 		});
 		http.formLogin(withDefaults());
 		http.httpBasic(withDefaults());
-		http.csrf((csrf) -> csrf.disable());
+		http.csrf(AbstractHttpConfigurer::disable);
```
Comment (wilkinsona) [inline spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-session-mongo/src/main/java/smoketest/session/mongodb/SecurityConfiguration.java:45]: I think `CsrfConfigurer::disable` would read better here.

File: spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-session-jdbc/src/main/java/smoketest/session/SecurityConfiguration.java [hunk 13]
```diff
@@ -41,7 +42,7 @@ SecurityFilterChain managementSecurityFilterChain(HttpSecurity http) throws Exce
 		});
 		http.formLogin(withDefaults());
 		http.httpBasic(withDefaults());
-		http.csrf((csrf) -> csrf.disable());
+		http.csrf(AbstractHttpConfigurer::disable);
```
Comment (wilkinsona) [inline spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-session-jdbc/src/main/java/smoketest/session/SecurityConfiguration.java:45]: I think `CsrfConfigurer::disable` would read better here.

File: spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-session-hazelcast/src/main/java/smoketest/session/hazelcast/SecurityConfiguration.java [hunk 14]
```diff
@@ -41,7 +42,7 @@ SecurityFilterChain managementSecurityFilterChain(HttpSecurity http) throws Exce
 		});
 		http.formLogin(withDefaults());
 		http.httpBasic(withDefaults());
-		http.csrf((csrf) -> csrf.disable());
+		http.csrf(AbstractHttpConfigurer::disable);
```
Comment (wilkinsona) [inline spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-session-hazelcast/src/main/java/smoketest/session/hazelcast/SecurityConfiguration.java:45]: I think `CsrfConfigurer::disable` would read better here.

File: spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-graphql/src/main/java/smoketest/graphql/SecurityConfig.java [hunk 15]
```diff
@@ -35,7 +36,7 @@ public class SecurityConfig {
 
 	@Bean
 	public DefaultSecurityFilterChain springWebFilterChain(HttpSecurity http) throws Exception {
-		return http.csrf((csrf) -> csrf.disable())
+		return http.csrf(AbstractHttpConfigurer::disable)
```
Comment (wilkinsona) [inline spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-graphql/src/main/java/smoketest/graphql/SecurityConfig.java:39]: I think `CsrfConfigurer::disable` would read better here.

File: spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-actuator/src/test/java/smoketest/actuator/ShutdownSampleActuatorApplicationTests.java [hunk 16]
```diff
@@ -74,7 +75,7 @@ static class SecurityConfiguration {
 
 		@Bean
 		SecurityFilterChain configure(HttpSecurity http) throws Exception {
-			http.csrf((csrf) -> csrf.disable());
+			http.csrf(AbstractHttpConfigurer::disable);
```
Comment (wilkinsona) [inline spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-actuator/src/test/java/smoketest/actuator/ShutdownSampleActuatorApplicationTests.java:78]: I think `CsrfConfigurer::disable` would read better here.

File: spring-boot-project/spring-boot-docs/src/main/java/org/springframework/boot/docs/data/sql/h2webconsole/springsecurity/DevProfileSecurityConfiguration.java [hunk 17]
```diff
@@ -35,8 +37,8 @@ public class DevProfileSecurityConfiguration {
 	SecurityFilterChain h2ConsoleSecurityFilterChain(HttpSecurity http) throws Exception {
 		http.securityMatcher(PathRequest.toH2Console());
 		http.authorizeHttpRequests(yourCustomAuthorization());
-		http.csrf((csrf) -> csrf.disable());
-		http.headers((headers) -> headers.frameOptions((frame) -> frame.sameOrigin()));
+		http.csrf(AbstractHttpConfigurer::disable);
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot-docs/src/main/java/org/springframework/boot/docs/data/sql/h2webconsole/springsecurity/DevProfileSecurityConfiguration.java:40]: I think `CsrfConfigurer::disable` would read better here.

File: spring-boot-project/spring-boot-devtools/src/main/java/org/springframework/boot/devtools/autoconfigure/RemoteDevtoolsSecurityConfiguration.java [hunk 18]
```diff
@@ -49,7 +50,7 @@ class RemoteDevtoolsSecurityConfiguration {
 	SecurityFilterChain devtoolsSecurityFilterChain(HttpSecurity http) throws Exception {
 		http.securityMatcher(new AntPathRequestMatcher(this.url));
 		http.authorizeHttpRequests((requests) -> requests.anyRequest().anonymous());
-		http.csrf((csrf) -> csrf.disable());
+		http.csrf(AbstractHttpConfigurer::disable);
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot-devtools/src/main/java/org/springframework/boot/devtools/autoconfigure/RemoteDevtoolsSecurityConfiguration.java:53]: I think `CsrfConfigurer::disable` would read better here.

File: spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/graphql/security/GraphQlWebMvcSecurityAutoConfigurationTests.java [hunk 19]
```diff
@@ -154,7 +155,7 @@ static class SecurityConfig {
 
 		@Bean
 		DefaultSecurityFilterChain springWebFilterChain(HttpSecurity http) throws Exception {
-			return http.csrf((c) -> c.disable())
+			return http.csrf(AbstractHttpConfigurer::disable)
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/graphql/security/GraphQlWebMvcSecurityAutoConfigurationTests.java:158]: I think `CsrfConfigurer::disable` would read better here.

**Review by wilkinsona — CHANGES_REQUESTED** [review]
Thanks for the updates, @asashour. I just noticed one other change that I think would improve things a little. Please see the comments for details.

**Comment by asashour** [comment]
If I may say something, there is an error for some time that `No space left on device`. I am not sure if this is related to the PR content.

**Comment by philwebb** [comment]
Thanks @asashour

**Pull request closed without merging**